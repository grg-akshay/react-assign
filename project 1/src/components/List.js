import React from 'react';
import '../styles/List.css';

const List = (props) => {
  return (
    <div className='list-container'>
        <div className='list-container__title'>EXISTING ARRAY</div>
        {

            props.existingList.length===0 
            ? 
            (<p>No item in array</p>)
            :
            props.existingList.map((item) =>{
                let classAssign = props.filteredList[item]===0 ?'skip ':''
                return <div key={item} className={classAssign} >{item}</div>
            })
        }
    </div>
  )
}

export default List
