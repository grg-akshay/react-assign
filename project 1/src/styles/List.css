.list-container{
  margin: 0 auto;
  margin-top: 60px;
  padding-top:10px;
  padding-bottom:10px;
  background:#316EA1;
  width: 500px;
  font-size: 60%;
  border-radius: 10px;
}

.skip{
    text-decoration: line-through;
}
.list-container__title{
    color: black;
    margin-bottom: 10px;
}