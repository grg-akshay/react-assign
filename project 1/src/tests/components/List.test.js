import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import List from '../../components/List';
import { existingList, listDefault } from '../fixtures/list';

test('should render <List /> correctly without search', () =>{
    const wrapper = shallow(<List 
                        existingList={existingList}
                        filteredList={listDefault}
                    />);
     expect(toJSON(wrapper)).toMatchSnapshot();
})

test('should render <List /> correctly with empty array', () =>{
    const wrapper = shallow(<List 
                        existingList={[]}
                        filteredList={listDefault}
                    />);
     expect(toJSON(wrapper)).toMatchSnapshot();
})