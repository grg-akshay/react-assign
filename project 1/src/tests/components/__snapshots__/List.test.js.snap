// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render <List /> correctly with empty array 1`] = `
<div
  className="list-container"
>
  <div
    className="list-container__title"
  >
    EXISTING ARRAY
  </div>
  <p>
    No item in array
  </p>
</div>
`;

exports[`should render <List /> correctly without search 1`] = `
<div
  className="list-container"
>
  <div
    className="list-container__title"
  >
    EXISTING ARRAY
  </div>
  <div
    className=""
    key="1000"
  >
    1000
  </div>
  <div
    className=""
    key="1002"
  >
    1002
  </div>
  <div
    className=""
    key="1004"
  >
    1004
  </div>
  <div
    className=""
    key="1006"
  >
    1006
  </div>
</div>
`;
